packages/squiggle-lang/src/ast/types.ts

```diff
@@ -178,6 +180,7 @@ type LetOrDefun = {
 type NodeLetStatement = N<
   "LetStatement",
   LetOrDefun & {
+    typeSignature: NodeTypeSignature;
```

Comment:
I think that would work, I figured it was bad practice to use null/undefined as an expected outcome which is why I created an explicit `ImplicitType` class. I could make it null instead. (Although the Squiggle style guide in the readme said to prefer `undefined` over `null`?)

Reply:
Yeah, I recommended nulls here because there are nulls in `NodeFloat` types, and consistency is more important than the style guide.

But I don't really remember why I used `null` instead of `undefined` in `NodeFloat`, and it doesn't matter much.